types/react/index.d.ts

```diff
@@ -209,7 +210,7 @@ declare namespace React {
         type: ClassType<P, ClassicComponent<P, ComponentState>, ClassicComponentClass<P>>): CFactory<P, ClassicComponent<P, ComponentState>>;
     function createFactory<P, T extends Component<P, ComponentState>, C extends ComponentClass<P>>(
         type: ClassType<P, T, C>): CFactory<P, T>;
-    function createFactory<P>(type: ComponentClass<P>): Factory<P>;
+    function createFactory<P>(type: ComponentType<P>): Factory<P>;
```

Comment:
I would actually use `JSXElementConstructor` instead, actually. It's more broad and doesn't require passing in function/class components that have the correct static properties).

Reply:
Hi! I tried to replace with `ComponentClass` with `JSXElementConstructor`.
Then I encounter the following error:

```
ERROR: 254:7  expect  TypeScript@next compile error: 
Type 'Factory<SCProps>' is not assignable to type 'FunctionComponentFactory<SCProps>'.
  Type 'ReactElement<SCProps, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<any, any, any>)>' is not assignable to type 'FunctionComponentElement<SCProps>'.
    Types of property 'type' are incompatible.
      Type 'string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<any, any, any>)' is not assignable to type 'FunctionComponent<SCProps>'.
        Type 'string' is not assignable to type 'FunctionComponent<SCProps>'.
ERROR: 259:7  expect  TypeScript@next compile error: 
Type 'Factory<SCProps>' is not assignable to type 'FunctionComponentFactory<SCProps>'.
```

I am not sure how to solve this problem. Any guess ?